compiler/index.cpp

```diff
@@ -195,36 +199,69 @@ void Index::fix_path(std::string &path) const {
 
 File *Index::get_file(std::string path) const {
   fix_path(path);
-  auto it = files.find(path);
-  return it == files.end() ? nullptr : it->second;
+  auto it = files_prev_launch.find(path);
+  if (it != files_prev_launch.end()) {
+    return it->second;
+  }
+  if (files_only_cur_launch.empty()) {
+    return nullptr;
+  }
+
+  std::lock_guard<std::mutex> guard{mutex_rw_cur_launch};
+  it = files_only_cur_launch.find(path);
+  if (it != files_only_cur_launch.end()) {
+    return it->second;
+  }
+  return nullptr;
 }
 
 File *Index::insert_file(std::string path) {
   fix_path(path);
-  File *&f = files[path];
-  if (f == nullptr) {
-    f = new File();
-    f->path = std::move(path);
-    kphp_assert (f->path.size() > get_dir().size());
-    kphp_assert (strncmp(f->path.c_str(), get_dir().c_str(), get_dir().size()) == 0);
-    f->name = vk::string_view{f->path}.substr(get_dir().size());
-    auto dot_i = f->name.rfind('.');
-    if (dot_i == string::npos) {
-      dot_i = f->name.size();
-    }
-    auto slash_i = f->name.rfind('/', dot_i - 1);
-    if (slash_i != string::npos) {
-      f->subdir = f->name.substr(0, slash_i + 1);
-      create_subdir(f->subdir);
-    } else {
-      f->subdir = vk::string_view{};
-    }
-    f->ext = f->name.substr(dot_i);
-    f->name_without_ext = f->name.substr(0, dot_i);
+
+  File *f = get_file(path);
+  if (f != nullptr) {
+    return f;
   }
+//  printf("%s not found in index, creating\n", file_name.c_str());
```

Comment:
Commented out code. Is it needed here?

Reply:
Just for potential logging/debugging in the future. We have some commented printfs throughout all our code. While not being the best practice, they point out, that this exact place is somehow important for logging.

Reply:
Thanks for the explanation, I will remember it for the future.